### ISSUE TITLE
Numeric object property

### ISSUE THREAD

**Issue by mightydes** [angus-c/just#139]
Good day!
What about case, when Integer used as Object property, for example:

```js
{
  c: 1,
  b: 2,
  a: 3,
  13: ['d1', 'd2', {i: 'ok'}]
  ^ problem is here
}
```
In diff `path` it looks like Array index...
It would be useful if the `pathConverter` method would have a second argument with types...

**Comment by angus-c** [comment]
Hi Lev,

Can you give a full example? (perhaps via the [REPL](https://anguscroll.com/just/just-diff))

**Comment by mightydes** [comment]
https://runkit.com/mightydes/5d108178ea5ab1001d95fcf4

![Выделение_129](https://user-images.githubusercontent.com/4075454/60001396-1044ee80-966f-11e9-994c-014d017d841f.png)

**Comment by angus-c** [comment]
I think this is correct, because array property indexes are actually strings. (JavaScript coerces number indexes into strings when assigning new values to an array.

```js
const a = ['aa'];
a[1] = 'bb'; // js coerces 1 to '1'
a['2'] = 'cc'; // no coercion necessary
a; // ['aa', 'bb', 'cc'];
```

**Comment by mightydes** [comment]
Yes, but it matters for assigning by path.

For example:

```js
// Current path behavior:
['a', 'b', '1', 'x']
            ^ can be Array index OR Object property

// Better path behavior:
['a', 'b', 1, 'x']
           ^ Number ie -- Array index
['a', 'b', '1', 'x']
            ^ String ie -- Object property
```

With current behavior you cannot acces path using [Ramda lens](https://ramdajs.com/docs/#lensPath) for example...

**Comment by angus-c** [comment]
Ok, I'll try to get to this in the next few days

**Comment by angus-c** [comment]
Fixed by #140 

just-diff@3.0.0
just-diff-apply@3.0.0
